crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -643,6 +644,17 @@ impl<'a> Resolver<'a> {
         }
     }
 
+    fn handle_contract_visibility(&mut self, func: &NoirFunction) -> ContractVisibility {
+        let mut contract_visibility = func.def.contract_visibility;
+        if !self.in_contract() && contract_visibility != ContractVisibility::Secret {
```

Comment:
It uses some assumptions about contract visibility, which would need to change if we change the behaviour. I think having an explicit visibility for a function will be more clear, like having an optional contract visibility (which is set only for contracts), or a normal function type.

Reply:
I've changed it to `Option<ContractVisibility>`